transport/http/handler.go

```diff
@@ -82,6 +87,26 @@ func (h handler) callHandler(w http.ResponseWriter, req *http.Request) error {
 
 	ctx := context.Background()
 
+	// Extract opentracing etc baggage from headers
+	// Annotate the inbound context with a trace span
+	tracer := h.Deps.Tracer()
+	carrier := opentracing.HTTPHeadersCarrier(req.Header)
+	parentSpanCtx, _ := tracer.Extract(opentracing.HTTPHeaders, carrier)
+	span := tracer.StartSpan(
+		treq.Procedure,
+		opentracing.StartTime(start),
+		opentracing.Tags{
+			"rpc.caller":   treq.Caller,
+			"rpc.service":  treq.Service,
+			"rpc.encoding": treq.Encoding,
+		},
+		ext.RPCServerOption(parentSpanCtx), // implies ChildOf
+	)
+	ext.PeerService.Set(span, treq.Caller)
+	// ext.HTTPUrl.Set(span, req.URL.String()) // XXX panics
+	defer span.Finish()
+	ctx = opentracing.ContextWithSpan(ctx, span)
```

Comment:
I would move this below the `ParseTTL` context. That one comes with a timeout while this one is off of background so no TTL.